src/core/meshcop/dataset_manager.cpp

```diff
@@ -638,6 +648,45 @@ Error DatasetManager::SendGetRequest(const Dataset::Components &aDatasetComponen
     return error;
 }
 
+void DatasetManager::HandleSetResponse(Coap::Message &aMessage, const Ip6::MessageInfo &aMessageInfo, Error aResult)
+{
+    Error    error;
+    StateTlv stateTlv;
+
+    OT_UNUSED_VARIABLE(aMessageInfo);
+
+    SuccessOrExit(error = aResult);
```

Comment:
`aResult` might be `kErrorResponseTimeout`?

Reply:
Yes. Added to `otDatasetMgmtSetCallback` documents.